src/RealMeService.php

```diff
@@ -636,6 +637,10 @@ public function getBackURL(HTTPRequest $request)
         if ($session->get('RealMeBackURL')) {
             $url = $session->get('RealMeBackURL');
             $session->clear('RealMeBackURL'); // Ensure we don't redirect back to the same error twice
+        } elseif ($request->requestVar('BackURL')) {
+            $url = $request->requestVar('BackURL');
+        } elseif ($request->postVar('RelayState')) {
```

Comment:
Is this needed to solve your issue? Or just trying to cover more ground?

Reply:
Yes, this is called in a number of different contexts.